### ISSUE TITLE
Dializer error when using UUID scalar recipe

### ISSUE THREAD

**Issue by datafoo** [absinthe-graphql/absinthe#694]
### Environment

* Elixir version (elixir -v):
```
$ elixir -v
Erlang/OTP 21 [erts-10.2.3] [source] [64-bit] [smp:3:3] [ds:3:3:10] [async-threads:1] [hipe]

Elixir 1.8.1 (compiled with Erlang/OTP 20)
```
* Absinthe version (mix deps | grep absinthe):
```
$ mix deps | grep absinthe
* absinthe 1.4.15 (Hex package) (mix)
  locked at 1.4.15 (absinthe) d9655f38
* absinthe_plug 1.4.6 (Hex package) (mix)
  locked at 1.4.6 (absinthe_plug) ac5d2d3d
```
* Dialyxir version:
```
$ mix deps | grep dialyxir
* dialyxir 0.5.1 (Hex package) (mix)
  locked at 0.5.1 (dialyxir) b331b091
```

### Expected behavior

Run `mix dializer` on my project should return no error.

### Actual behavior

Running `mix dializer` on my project using the [UUID recipe](https://github.com/absinthe-graphql/absinthe/wiki/Scalar-Recipes#uuid-using-ectouuid) returns:
```
:0: Unknown type 'Elixir.Result.Object':t/0
```
I noticed that commenting out the 2 lines starting with `@spec` in the recipe would not trigger the dializer error.

I also noticed that the [`Absinthe.Blueprint.Execution`](https://github.com/absinthe-graphql/absinthe/blob/master/lib/absinthe/blueprint/execution.ex#L48) module refers to module `Result.Object`. Was it meant to be [`Absinthe.Blueprint.Result.Object`](https://github.com/absinthe-graphql/absinthe/blob/master/lib/absinthe/blueprint/result/object.ex) instead?

**Comment by beardedeagle** [comment]
Interesting. I'll have to take a look.

**Comment by ukrbublik** [comment]
Can this be merged to 1.4 branch?
Or if there is workaround to ignore this in dialyzer report?

**Comment by benwilson512** [comment]
@ukrbublik 1.4 is only receiving critical bug fixes at this point since 1.5 is in RC.

**Comment by ukrbublik** [comment]
Ok, thanks. Workaround is just put
```
:0: Unknown type 'Elixir.Result.Object':t/0
```
in dialyzer's `ignore_warnings` confg file
